firmware/Src/synth/Osc.h

```diff
@@ -152,14 +166,18 @@ class Osc : public SynthStateAware
             localLastValue0 = newValue - localLastValue1 + .99525f * localLastValue0;
             localLastValue1 = newValue;
 
-            oscValuesToFill[k] = localLastValue0 * freqMultiplier * localEnv ;
-            localEnv += envInc;
+            oscValuesToFill[k] = localLastValue0 * localEnvM ;
+            localEnvM += envIncM;
+
+            //oscValuesToFill[k] = localLastValue0 * freqMultiplier * localEnv ;
+            //localEnv += envInc;
 
         }
         lastValue[0] = localLastValue0;
         lastValue[1] = localLastValue1;
 
-        env = localEnv;
+        env = localEnv + envInc * 32;

```

Comment:
@pvig can you explain this change ?

Reply:
Sure, it's an optimisation, to save one multiply in Osc.h ( 32 * actually)
in getNextBlockWithFeedbackAndEnveloppe instead of 
`oscValuesToFill[k] = localLastValue0 * freqMultiplier * localEnv ;`
we have 
`oscValuesToFill[k] = localLastValue0 * localEnvM ;`

after the loop, `env = localEnv + envInc * 32; ` set env to the right value because localEnv is no more incremented in the loop.

The whole optimisation is first commited here :
https://github.com/pvig/preenfm3/commit/994b45f5296d9bfe673836f2a6e2452c2fadc168#diff-3414c293763eb76ea7f5983f4fd0425e137d409a86baa49c46fe0e8c767f7b86

it may not be really significant though, but why not ? ^^